checker/tests/upperbound/Constants.java

```diff
@@ -0,0 +1,10 @@
+
+class Constants {
+
+    void test() {
+        int[] arr = {1, 2, 3};
+        int i = arr[1];
+        //:: warning: (array.access.unsafe.high)
```

Comment:
This is the test that actually tests array accesses.  Could you please add some more? I also don't see any test for refining a variable used as an array dimension.